[new product] chef-infra-server (#4425)
captn3m0/endoflife.date: products/chef-infra-server.md

---
title: Chef Infra Server
category: server-app
iconSlug: chef
tags: erlang-runtime ruby-runtime

> permalink: /chefinfraserver

Reviewer: ```suggestion
permalink: /chef-infra-server
```

Just like the markdown file ?

alternate_urls:

> -   /chefserver

Reviewer: ```suggestion
-   /chef-server
```

WDYT ?

versionCommand: chef-server-ctl version
releasePolicyLink: "https://docs.chef.io/versions/"
changelogTemplate: "https://docs.chef.io/release_notes_server/#__LATEST__"

> 

Reviewer: ```suggestion
```

formatting

activeSupportColumn: true
releaseDateColumn: true

> auto:

Reviewer: Can you open a PR on https://github.com/endoflife-date/release-data so we can merge both at the same time ?

Author: done

- custom: true

# eol(x) = releaseDate(x+2) > confirm EOL date here https://docs.chef.io/versions/
# support(x) = releaseDate(x+1)
releases:
-   releaseCycle: "15"
    releaseDate: 2022-06-13
    support: true
    eol: false

>     latest: "15.9.20"

Reviewer: No `latestReleaseDate` ?

Author: updated with the custom script

-   releaseCycle: "14"
    releaseDate: 2020-06-08
    support: 2022-06-13
    eol: false
    latest: "14.16.19"

-   releaseCycle: "13"
    releaseDate: 2019-05-31
    support: 2020-06-08
    eol: 2021-06-30
    latest: "13.2.0"

-   releaseCycle: "12"
    releaseDate: 2014-11-24
    support: 2019-05-31
    eol: 2020-12-31
    latest: "12.19.31"

-   releaseCycle: "11"
    releaseDate: 2013-09-30
    support: 2014-11-24
    eol: 2018-12-31
    latest: "11.1.7"

---

> [Chef Infra Server](https://docs.chef.io/server/) is an open-source platform that complements Chef Infra Client.
> It serves as a central hub for configuration data, providing scalable infrastructure management. 

Chef Infra Server generally follows an N-1 support strategy  

> (meanings that the 2 most recent major versions receive security and bug fixes)

Reviewer: Could you give a bit more details on the two support phases explained on https://docs.chef.io/versions/#lifecycle-definitions (example of such documentation: https://endoflife.date/nodejs) ? I quickly went through https://docs.chef.io/versions/#lifecycle-definitions and I did not find the rule `support(x) = releaseDate(x+1)` (e.g. when a release become deprecated).

Author: Based on [Here](https://www.chef.io/service-level-agreement?_gl=1*1xlphvy*_ga*ODEyNjY0NTc4LjE3MDM3MTUwMDg.*_ga_9JSNBCSF54*MTcwODQ0OTk4Mi4xNS4xLjE3MDg0NTI2MzcuNTkuMC4w*_gcl_au*MTYyMjE3NTI1MS4xNzAzNzE1MDk4)

“Deprecated” means Supported Applications that are supported by Chef but for which a newer version or alternative Supported Applications are available and recommended for Chef customer use. Chef no longer provides scheduled releases for Deprecated Software but may produce ad hoc Updates in response to customer issues and/or security vulnerabilities at the sole discretion of Chef. Deprecated Software is targeted for End of Life but a timeline has not yet been announced for End of Life.

And here, https://docs.chef.io/versions/#deprecated-products-and-versions

v15 is GA
v14 is "deprecated"
V13 and below are EOL

Based on that, it seems coherent to me, but maybe I don't understand where you're going with this?
```
 eol(x) = releaseDate(x+2) > confirm EOL date here https://docs.chef.io/versions/
 support(x) = releaseDate(x+1)
```

I updated the desc as follow, let me know.

The only actively maintained version [is the latest](https://docs.chef.io/versions/#supported-commercial-distributions). Release N-1 are "deprecated" [and can only receive security updates](https://www.chef.io/service-level-agreement?_gl=1*1xlphvy*_ga*ODEyNjY0NTc4LjE3MDM3MTUwMDg.*_ga_9JSNBCSF54*MTcwODQ0OTk4Mi4xNS4xLjE3MDg0NTI2MzcuNTkuMC4w*_gcl_au*MTYyMjE3NTI1MS4xNzAzNzE1MDk4). Releases N-2 [and below are EOL](https://docs.chef.io/versions/#end-of-life-eol-products).

---
title: Chef Infra Server
category: server-app
tags: erlang-runtime ruby-runtime
iconSlug: chef
permalink: /chef-infra-server
alternate_urls:
-   /chef-server
versionCommand: chef-server-ctl version
releasePolicyLink: https://docs.chef.io/versions/
changelogTemplate: https://docs.chef.io/release_notes_server/#__LATEST__
activeSupportColumn: true
releaseDateColumn: true

identifiers:
-   repology: chef-infra-server

auto:
  methods:
  -   custom: chef-infra-server

# support(x) = releaseDate(x+1)
# eol(x) = releaseDate(x+2) or the date documented on https://docs.chef.io/versions/
releases:
-   releaseCycle: "15"
    releaseDate: 2022-06-13
    support: true
    eol: false
    latest: "15.9.20"
    latestReleaseDate: 2024-01-10

-   releaseCycle: "14"
    releaseDate: 2020-06-08
    support: 2022-06-13
    eol: false
    latest: "14.16.19"
    latestReleaseDate: 2022-06-07

-   releaseCycle: "13"
    releaseDate: 2019-05-31
    support: 2020-06-08

>     eol: 2021-06-30

Reviewer: ```suggestion
    eol: 2021-06-30 # https://web.archive.org/web/20210919232343/https://docs.chef.io/versions/
```

Reviewer 2: If you're looking for the source fo this date I wrote the blog on it back in 2021: https://www.chef.io/blog/announcing-the-chef-infra-server-13-deprecation

Reviewer: Thanks !

    latest: "13.2.0"
    latestReleaseDate: 2020-04-10

-   releaseCycle: "12"

>     releaseDate: 2014-11-24

Reviewer: ```suggestion
    releaseDate: 2014-11-25
```

See https://github.com/chef/chef-server/releases/tag/12.0.0.

    support: 2019-05-31

>     eol: 2020-12-31

Reviewer: ```suggestion
    eol: 2020-12-31 # https://web.archive.org/web/20210613113759/https://docs.chef.io/versions/
```

    latest: "12.19.31"
    latestReleaseDate: 2019-03-07

-   releaseCycle: "11"

>     releaseDate: 2013-09-30

Reviewer: Where can I confirm this date ?

Author: release date was wrong, I updated

>     support: 2014-11-24

Reviewer: ```suggestion
    support: 2014-11-25
```

See https://github.com/chef/chef-server/releases/tag/12.0.0.

>     eol: 2018-12-31

Reviewer: Why not 2019-05-31 (the release date of 13) ?

Author: https://www.chef.io/blog/end-of-life-announcement-for-chef-reporting-enterprise-chef-server-11-and-chef-analytics

    latest: "11.1.7"
    latestReleaseDate: 2014-06-19

---

> [Chef Infra Server](https://docs.chef.io/server/) is an open-source platform that complements
> Chef Infra Client. It serves as a central hub for configuration data, providing scalable
> infrastructure management.

> The only actively maintained version [is the latest](https://docs.chef.io/versions/#supported-commercial-distributions).
> Release N-1 are "deprecated" [and can only receive security updates](https://www.chef.io/service-level-agreement).
> Releases N-2 [and below are EOL](https://docs.chef.io/versions/#end-of-life-eol-products).

Reviewer: ```suggestion
Supported releases of Chef Infra Server are documented on [Chef's website](https://docs.chef.io/versions/#supported-commercial-distributions).
Looking at this document it seems that Chef Infra Server follows an N-1 support strategy, with the
latest release supported with bug and security fixes, and the previous release marked "deprecated"
and only supported with security fixes.
```

WDYT ?

---
title: Chef Infra Server
category: server-app
tags: erlang-runtime ruby-runtime
iconSlug: chef
permalink: /chef-infra-server
alternate_urls:
-   /chef-server
versionCommand: chef-server-ctl version
releasePolicyLink: https://docs.chef.io/versions/
changelogTemplate: https://docs.chef.io/release_notes_server/#__LATEST__
activeSupportColumn: true
releaseDateColumn: true

identifiers:
-   repology: chef-infra-server

auto:
  methods:
  -   custom: chef-infra-server

# support(x) = releaseDate(x+1)
# eol(x) = releaseDate(x+2) or the date documented on https://docs.chef.io/versions/
releases:
-   releaseCycle: "15"
    releaseDate: 2022-06-13
    support: true
    eol: false
    latest: "15.9.20"
    latestReleaseDate: 2024-01-10

-   releaseCycle: "14"
    releaseDate: 2020-06-08
    support: 2022-06-13
    eol: false
    latest: "14.16.19"
    latestReleaseDate: 2022-06-07

-   releaseCycle: "13"
    releaseDate: 2019-05-31
    support: 2020-06-08

>     eol: 2021-06-30 # https://web.archive.org/web/20210919232343/https://docs.chef.io/versions/

Reviewer: ```suggestion
    eol: 2021-06-30 # https://www.chef.io/blog/announcing-the-chef-infra-server-13-deprecation
```

Thanks @tas50 !

---
title: Chef Infra Server
category: server-app
iconSlug: chef
tags: erlang-runtime ruby-runtime
permalink: /chefinfraserver
alternate_urls:
-   /chefserver
releasePolicyLink: "https://docs.chef.io/versions/"
changelogTemplate: "https://docs.chef.io/release_notes_server/#__LATEST__/"
versionCommand: chef-server-ctl version
releaseDateColumn: true
activeSupportColumn: true

auto:

> -   git: https://github.com/chef/chef-server

Reviewer 2: The repository doesn't represent the actual versions. Each PR that is merged automatically bumps the patch release, minor release, or major release depending on GitHub labels that are applied. When the team is ready they promote these versions so you end up with versions like 1.0.15 and 1.1.24 being the released versions. There is an API called omnitruck that will tell you stable release versions.

https://docs.chef.io/api_omnitruck/

Author: Thank you for your feedback, I didn't have this information. I [quickly looked](https://omnitruck.chef.io/stable/chef/metadata?p=debian&pv=11&m=x86_64) at Omnibus, I'm afraid that the latest versions differ depending on the OS & platform

I think it would be best and the simplest to get the latest version from the [product release page](https://docs.chef.io/release_notes_server/#15.9.20), WDYT?

I have the impression that there is a release note for each version.   
Based on this, we could use this script to update the data.

```python
import requests
from bs4 import BeautifulSoup

url = "https://docs.chef.io/release_notes_server/"
response = requests.get(url)

if response.status_code == 200:
    soup = BeautifulSoup(response.text, 'html.parser')
    h2_elements = soup.find_all('h2', text=lambda text: "Chef Infra Server" in text)
    for h2_element in h2_elements:
        version_id = h2_element.get('id')
        print(f"{version_id}")
else:
    print(f"{response.status_code}")
```

Output :
```
18.3.0
18.2.7
18.1.0
18.0.185
18.0.155
17.10.95
17.10.3
17.10.0
...
```

In both cases it is impossible to retrieve the release date of the latest versions, so I removed the field.

Reviewer: Unfortunately the date is not documented on https://docs.chef.io/release_notes_server/, so you can't use this alone to add some automation. Maybe by combining both this and information from https://github.com/chef/chef-server/tags ?

Author: Good idea, I hadn't thought of that, I did that and it works, I'm going to open a PR on the other repo

    # https://regex101.com/r/ZBVO7U/1
    regex: ^(?<major>\d+)\.?(?<minor>\d{1,3})\.?(?<patch>\d+)\.?(?<tiny>\d+)?$

# eol(x) = releaseDate(x+2) > confirm EOL date here https://docs.chef.io/versions/
# support(x) = releaseDate(x+1)
releases:

> # -   releaseCycle: "16"

Reviewer 2: V16 should just get removed since that's not even branched at this point

Author: removed, ty

#    releaseDate: 2023-11-26
#    support: true
#    eol: false
#    latest: "16.0.0"

> #    latestReleaseDate: 2023-11-26

Author: v16 is available on github but I haven't found any announcement anywhere.
And it's not even on the official page, I think the best for now is to leave it commented

I don't know what to think about it :/
Apart from that, LGTM

Reviewer 2: Chef uses a tool called Expeditor that they wrote which handles major version bumping. It looks like November someone accidently applied the GitHub label to a PR which bumps the major version. They then reverted it in a followup commit. The versions in the VERSION file on main are not the actual released versions. You need to check what ends up on the docs site or what goes into the discourse. 16.0 should get removed here. It's probably going to be a while before they do a major version bump.  They did release a new minor version today though so this PR should get updated for that:

https://discourse.chef.io/t/chef-infra-server-15-9-20-released/22743

Author: removed & updated, ty
